src/formatDate.js

```diff
@@ -51,6 +51,43 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const newDate = [];
+  const oldDate = date.split(fromFormat[fromFormat.length - 1]);
+
+  function findSameElement(index) {
+    for (const oldElement of fromFormat) {
+      const indexOfOld = fromFormat.indexOf(oldElement);
+      let newElement = oldDate[indexOfOld];
+
+      if (oldElement === toFormat[index]) {
+        newDate.push(newElement);
+        break;
+      }
+
+      const isBothYears = oldElement.includes('Y')
+      && toFormat[index].includes('Y');
+      const isNewLengthBigger = oldElement.length < toFormat[index].length;
+
+      if (isBothYears && !isNewLengthBigger) {
+        newElement = oldDate[indexOfOld].slice(2);
+        newDate.push(newElement);
+      }
```

Comment:
oldDate[indexOfOld] already in newElement, perhaps it will be more clear to change newElement

`newElement = newElement.slice(2)` or `newDate.push(newElement.slice(2))`